flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/optimize/program/DynamicPartitionPruningProgramTest.java

```diff
@@ -658,4 +658,47 @@ public void testDPPWithJoinKeysNotIncludePartitionKeys() {
                         + " and dim.price < 500 and dim.price > 300";
         util.verifyRelPlan(query);
     }
+
+    @Test
+    public void testDppFactSideCannotReuseWithSameCommonSource() {
+        String query =
+                "SELECT * FROM(\n"
+                        + " Select fact_part.id, fact_part.price, fact_part.amount from fact_part join (Select * from dim) t1"
+                        + " on fact_part.fact_date_sk = dim_date_sk where t1.price < 500\n"
+                        + " UNION ALL Select fact_part.id, fact_part.price, fact_part.amount from fact_part)";
+        util.verifyExecPlan(query);
+    }
+
+    @Test
+    public void testDimSideReuseWithUnionAllTwoFactSide() {
+        util.tableEnv()
+                .executeSql(
```

Comment:
Can this test be simplified? I feel that the correlation between testing and this pr is not strong.